packages/datadog-instrumentations/src/next.js

```diff
@@ -147,18 +199,59 @@ function finish (ctx, result, err) {
   return result
 }
 
+// transpiled functions in newer next.js versions can't be shimmed
+// because they're read-only, so we re-define the property needed here
+function wrapObjectProperty (module, func, wrapper) {
+  const exported = Object.getOwnPropertyDescriptors(module)
+  const original = exported[func].get()
+  return Object.defineProperty(exported, func, {
+    enumerable: true,
+    get: function () { return wrapper(original) }
+  })
+}
+
+addHook({
+  name: 'next',
+  versions: ['>=13.4.13'],
+  file: 'dist/server/lib/router-utils/resolve-routes.js'
+}, resolveRoutesModule => wrapObjectProperty(resolveRoutesModule, 'getResolveRoutes', wrapGetResolveRoutes))
+
+addHook({
+  name: 'next',
+  versions: ['13.4.13'],
+  file: 'dist/server/lib/setup-server-worker.js'
+}, setupServerWorker => wrapObjectProperty(setupServerWorker, 'initializeServerWorker', wrapSetupServerWorker))
+
+addHook({
+  name: 'next',
+  versions: ['>=13.4.15'],
+  file: 'dist/server/lib/router-server.js'
+}, routerServer => wrapObjectProperty(routerServer, 'initialize', wrapInitialize))
+
 addHook({ name: 'next', versions: ['>=13.2'], file: 'dist/server/next-server.js' }, nextServer => {
   const Server = nextServer.default
 
-  shimmer.wrap(Server.prototype, 'handleRequest', wrapHandleRequest)
-  shimmer.wrap(Server.prototype, 'handleApiRequest', wrapHandleApiRequestWithMatch)
   shimmer.wrap(Server.prototype, 'renderToResponse', wrapRenderToResponse)
   shimmer.wrap(Server.prototype, 'renderErrorToResponse', wrapRenderErrorToResponse)
   shimmer.wrap(Server.prototype, 'findPageComponents', wrapFindPageComponents)
 
   return nextServer
 })
 
+addHook({
+  name: 'next',
+  versions: ['>=13.2 <13.4.13', '>=13.4.15'], // all versions but 13.4.13
```

Comment:
Wouldn't this also skip `13.4.14`?

Reply:
There isn't a released `13.4.14` version (trying to install it makes you choose from a list of available versions). I still tried to make this range include `13.4.14` anyways, but our tests failed when trying to run `yarn services` and not finding `13.4.14` in the npm registry

Reply:
Ah, that's right. Can you update the comment to be explicit about this? Seems like the kind of thing that will trip us up in the future.